frontend/app/components/DateSelect/index.tsx

```diff
@@ -0,0 +1,87 @@
+import type { ReactElement } from "react";
+import { CalendarIcon } from "@heroicons/react/24/outline";
+import ReactDatePicker from "react-datepicker";
+import { useRange } from "react-instantsearch";
+import { useState, useEffect } from "react";
+
+import "react-datepicker/dist/react-datepicker.css";
+
+const DatePicker =
+  (ReactDatePicker as unknown as { default: typeof ReactDatePicker }).default ??
+  ReactDatePicker;
+
+const DateSelect = (): ReactElement => {
+  const { range, refine, start } = useRange({
+    attribute: "endDateTimestamp",
+  });
+
+  const { min, max } = range;
+
+  const [from, setFrom] = useState(min);
+  const [to, setTo] = useState(max);
+
+  const refineEndDate = (fromDate: Date | null, toDate: Date | null) => {
+    // Convert to Unix timestamp before refining
+    const min = fromDate ? fromDate.getTime() : undefined;
+    const max = toDate ? toDate.getTime() : undefined;
+    refine([min, max]);
+  };
+
+  const handleFromDateChange = (date: Date | null) => {
+    if (to && date && date.getTime() > to) {
```

Comment:
Looks like this and the code below in `handleToDateChange` are protecting from the dates going past the range in either direction, yeah?

Reply:
Exactly